addOns/soap/src/main/java/org/zaproxy/zap/extension/soap/ExtensionImportWSDL.java

```diff
@@ -164,6 +183,37 @@ public void fileUrlWSDLImport(final File file) {
         parser.extFileWSDLImport(file, THREAD_PREFIX + threadId++);
     }
 
+    private void addScript() throws IOException {
+        ExtensionScript extScript =
+                Control.getSingleton().getExtensionLoader().getExtension(ExtensionScript.class);
+        String scriptName = "SOAP Support.js";
+        if (extScript != null && extScript.getScript(scriptName) == null) {
+            ScriptType variantType =
+                    extScript.getScriptType(ExtensionActiveScan.SCRIPT_TYPE_VARIANT);
+            ScriptEngineWrapper engine = extScript.getEngineWrapper("Oracle Nashorn");
```

Comment:
@thc202 @psiinon do we need to think of a more generic way to do this? In a few Java revisions nashorn won't be available and we might be dependent on graalvm or something else. Should the script extension have a way to stating engines for languages?

Reply:
It's more reliable to use a specific engine as there are no guarantees that the other engines will support what the script is doing (while GraalVM JavaScript provides compatibility with Nashorn it might not work always).

(Latest Java version no longer include Nashorn.)